image_tools/src/cam2image.cpp

```diff
@@ -140,11 +138,8 @@ int main(int argc, char * argv[])
     };
 
   // Set the QoS profile for the subscription to the flip message.
-  rmw_qos_profile_t custom_flip_qos_profile = rmw_qos_profile_sensor_data;
-  custom_flip_qos_profile.depth = 10;
-
   auto sub = node->create_subscription<std_msgs::msg::Bool>(
-    "flip_image", callback, custom_flip_qos_profile);
+    "flip_image", rclcpp::SensorDataQoS(), callback);
```

Comment:
This is more related with the main PR (https://github.com/ros2/rclcpp/pull/713) that here. But if the idea is to encourage users to always specify history depth, shouldn't this looks like:

```cpp
auto sub = node->create_subscription<std_msgs::msg::Bool>(
    "flip_image", rclcpp::SensorDataQoS(5), callback);
```

Or, is fine having some default profiles with an implicit history depth?

Reply:
Yeah, so this one, is something worth discussing, I'll probably bring it up at the ROS 2 meeting if there's time.

Basically, for profiles that use keep last, requiring them to specify the depth, kind of makes sense, for the same reason we ask people to do this for the normal case, however, if the existing rmw profile uses keep all then it doesn't make sense to require a depth. We could require a `QoSInitialization` which is either keep all or keep last + a depth, but the reason I didn't require that is if a profile were to change from defaulting from keep last to keep all, then the user would be unintentionally overriding this setting always. The effect would be that no rmw profile's history settings would ever be used as the default, making them kind of meaningless.

Maybe we want to go that way, but I'm not sure.

Reply:
Based on the discussion in our meeting, we think in response to @ivanpauno's question:

> Or, is fine having some default profiles with an implicit history depth?

Yes, especially for cases like `rclcpp::ServicesQoS`, but also for less obvious cases like `rclcpp::SensorDataQoS`.

Therefore, I'll leave this as-is.